src/subdomains/supporting/payment/services/transaction.service.ts

```diff
@@ -10,37 +15,45 @@ import { TransactionRepository } from '../repositories/transaction.repository';
 export class TransactionService {
   constructor(private readonly repo: TransactionRepository) {}
 
-  async create(dto: CreateTransactionDto): Promise<Transaction | undefined> {
+  async create(
+    dto: CreateTransactionDto,
+    sourceEntity: BankTx | CryptoInput | CheckoutTx | RefReward,
+  ): Promise<Transaction | undefined> {
     const entity = this.repo.create(dto);
 
+    const hash = Util.createHash(
+      `${sourceEntity.id}` + entity.created + entity.sourceType + sourceEntity.created,
+    ).toUpperCase();
```

Comment:
Does the entity already have a `created` date? I think I would take only fields from source entity.